File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        // 완주하지 못한 선수의 이름을 return
        // 동명이인 존재
        String answer = "";
        Map<String,Integer> map = new HashMap<>();

        for(int i=0; i<participant.length; i++){
            map.put(participant[i],map.getOrDefault(participant[i],0)+1);
        }
        
        for(int i=0; i<completion.length; i++){
            map.put(completion[i],map.get(completion[i])-1);
            
            if(map.get(completion[i]) == 0){
                map.remove(completion[i]);
            }
        }
        
        for(String name: map.keySet()){
            answer = name;
        }
   
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    # 참여 선수들 이름이 담긴 배열 participant
    # 완주한 선수들 이름이 담긴 배열 completion
    # 완주하지 못한 선수의 이름을 return 해라
    
    my_dict = {}
    
    
    for player in participant :
        if player in my_dict:
            my_dict[player] +=1
        else : my_dict[player] =1
        
    for player in completion :
        my_dict[player] -=1
        
    for player in my_dict :
        if my_dict[player] ==1:
            return player
        
